ntoskrnl/io/iomgr/irq.c

```diff
@@ -166,4 +166,67 @@ IoDisconnectInterrupt(PKINTERRUPT InterruptObject)
     ExFreePool(IoInterrupt); // ExFreePoolWithTag(IoInterrupt, TAG_KINTERRUPT);
 }
 
+#if (NTDDI_VERSION >= NTDDI_LONGHORN) || defined(__REACTOS__)
+NTSTATUS
+NTAPI
+IopConnectInterruptEx_FullySpecific(_Inout_ PIO_CONNECT_INTERRUPT_PARAMETERS Parameters)
+{
+    NTSTATUS Status;
+    IO_CONNECT_INTERRUPT_PARAMETERS LocParameters;
+    PAGED_CODE();
+
+    LocParameters = *Parameters;
+
+    /* Only fully FullySpecified interrupts this can call the legacy routine */
+    Status = IoConnectInterrupt(LocParameters.FullySpecified.InterruptObject,
+                                LocParameters.FullySpecified.ServiceRoutine,
+                                LocParameters.FullySpecified.ServiceContext,
+                                LocParameters.FullySpecified.SpinLock,
+                                LocParameters.FullySpecified.Vector,
+                                LocParameters.FullySpecified.Irql,
+                                LocParameters.FullySpecified.SynchronizeIrql,
+                                LocParameters.FullySpecified.InterruptMode,
+                                LocParameters.FullySpecified.ShareVector,
+                                LocParameters.FullySpecified.ProcessorEnableMask,
+                                LocParameters.FullySpecified.FloatingSave);
+    DPRINT("IopConnectInterruptEx_FullySpecific: has failed with status %X", Status);
+    return Status;
+}
+
+NTSTATUS
+NTAPI
+IoConnectInterruptEx(_Inout_ PIO_CONNECT_INTERRUPT_PARAMETERS Parameters)
```

Comment:
```suggestion
IoConnectInterruptEx(
    _Inout_ PIO_CONNECT_INTERRUPT_PARAMETERS Parameters)
```

Reply:
(NOTE: For this function and the two others introduced there.)